dex/networks/eth/tokens.go

```diff
@@ -0,0 +1,145 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+//go:build lgpl
+// +build lgpl
+
+package eth
+
+import (
+	"fmt"
+	"math/big"
+
+	"decred.org/dcrdex/dex"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Token is the definition of an ERC20 token, including all of its network and
+// version variants.
+type Token struct {
+	*dex.Token
+	// NetTokens is a mapping of token addresses for each network available.
+	NetTokens map[dex.Network]*NetToken `json:"netAddrs"`
+	// EVMFactor allows for arbitrary ERC20 decimals. For an ERC20 contract,
+	// the relation
+	//    math.Log10(UnitInfo.Conventional.ConversionFactor) + Token.EVMFactor = decimals
+	// should hold true.
+	// Since most assets will use a value of 9 here, a default value of 9 will
+	// be used in AtomicToEVM and EVMToAtomic if EVMFactor is not set.
+	EVMFactor *int64 `json:"evmFactor"` // default 9
```

Comment:
I've read the discussion you had with @martonp on this, but I'm still a bit fuzzy on this.
`math.Log10(UnitInfo.Conventional.ConversionFactor) + Token.EVMFactor = decimals`
Is 9 still an expected `EVMFactor` default?  Sounded like you would have 6 and 0?

Take the `testTokenID`, defined below.  It has `ConversionFactor` of 1e8 and `nil` for `EVMFactor`.  The `factor()` method would return 10^9.  But log10(conversionFactor) is 8, so that makes `decimals` get computed as ... 17?

Reply:
The `ConversionFactor` for `testTokenID` is `GweiFactor = 1e9`, so decimals is `18`, which is the default for a token if not specified.